synapse/storage/databases/main/events_worker.py

```diff
@@ -657,10 +660,18 @@ async def get_missing_events_from_db() -> Dict[str, EventCacheEntry]:
                 # the events have been redacted, and if so pulling the redaction event
                 # out of the database to check it.
                 #
+                missing_events = {}
                 try:
-                    missing_events = await self._get_events_from_db(
+                    # First fetch from the cache - including any external caches
+                    cache_missing_events = await self._get_events_from_cache(
                         missing_events_ids,
                     )
```

Comment:
again this feels like redundant work on non-external-cache systems, and double-counting of metrics in any case.